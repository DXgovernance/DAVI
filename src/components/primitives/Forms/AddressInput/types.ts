import { InputProps } from 'components/primitives/Forms/Input';

export interface AddressInputProps extends InputProps<string> {
  enableENSName?: boolean;
}
