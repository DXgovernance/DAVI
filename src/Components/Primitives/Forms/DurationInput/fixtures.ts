export const defaultProps = {
  value: 0,
  onChange: () => jest.fn(),
};
