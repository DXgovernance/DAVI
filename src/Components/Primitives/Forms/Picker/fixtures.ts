import { PickerProps } from './types';

export const fullProps: PickerProps = {
  data: [
    {
      id: '0',
      title: 'Different',
      subtitle: 'There are no references here',
      value: '0',
    },
    {
      id: '1',
      title: 'Token1',
      subtitle: 'This is the token one',
      value: 'value token 1',
      balance: 135,
    },
    {
      id: '2',
      title: 'Token2',
      value: 'value token 2',
    },
    {
      id: '3',
      title: 'Token3',
      subtitle: 'Some subtitle',
      value: 'value token 3',
    },
    {
      id: '4',
      title: 'Token4',
      value: 'value token 4',
    },
    {
      id: '5',
      title: 'Token5',
      value: 'value token 5',
      balance: 350,
    },
    {
      id: '6',
      title: 'Token6',
      subtitle: 'Indeed this is sub-text',
      value: 'value token 6',
    },
  ],
  header: 'Pick a token',
  isOpen: true,
  onSelect: option => console.log(option),
  onClose: () => {},
};

export const swaprPairs: PickerProps = {
  isOpen: true,
  onSelect: option => console.log(option),
  onClose: () => {},
  header: 'Pick a pair',
  data: [
    {
      title: 'TORN-WETH',
      address: '0x010946a9189f4fa8d6bf01d2fa32e0edf3d6e03b',
      reserve0: '21.979143344333184701',
      reserve1: '0.295076965839191706',
      reserveUSD: '569.2364021717024122251006928140325',
      reserveNativeCurrency: '0.295076965839191706',
      totalSupply: '2.499685243646281452',
      token0: {
        address: '0x77777feddddffc19ff86db637967013e6c6a116c',
        name: 'TornadoCash',
        symbol: 'TORN',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'STAKE-DXD',
      address: '0x04e9644c7b5c82280fe95f515b7c4ad18a72db83',
      reserve0: '0.321291693076585919',
      reserve1: '0.020502751836012276',
      reserveUSD: '15.32220501881621236926362444314866',
      reserveNativeCurrency: '0.007406919217920272769253686646761754',
      totalSupply: '0.080549044622494839',
      token0: {
        address: '0x0ae055097c6d159879521c384f1d2123d1f195e6',
        name: 'STAKE',
        symbol: 'STAKE',
        decimals: '18',
      },
      token1: {
        address: '0xa1d65e8fb6e87b60feccbc582f7f97804b725521',
        name: 'DXdao',
        symbol: 'DXD',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'DAI-USDC',
      address: '0x0e18852ee5ce266e2ce6f4844ba04cd9cd11af5b',
      reserve0: '0.000083384608530149',
      reserve1: '0.000084',
      reserveUSD: '0.0001632092461365910476688685934587097',
      reserveNativeCurrency: '0.000000264969183078724715385506512724458',
      totalSupply: '0.000000000083420641',
      token0: {
        address: '0x6b175474e89094c44da98b954eedeac495271d0f',
        name: 'Dai Stablecoin',
        symbol: 'DAI',
        decimals: '18',
      },
      token1: {
        address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
        name: 'USD Coin',
        symbol: 'USDC',
        decimals: '6',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'TCAP-CTX',
      address: '0x15d37a2c7e4a88adb0ef5a6c85866a54d261201d',
      reserve0: '0.004363640935513935',
      reserve1: '0.087746620703447774',
      reserveUSD: '0',
      reserveNativeCurrency: '0',
      totalSupply: '0.019496932818319274',
      token0: {
        address: '0x16c52ceece2ed57dad87319d91b5e3637d50afa4',
        name: 'TCAP Token',
        symbol: 'TCAP',
        decimals: '18',
      },
      token1: {
        address: '0x321c2fe4446c7c963dc41dd58879af648838f98d',
        name: 'Cryptex',
        symbol: 'CTX',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'USDC-CRV',
      address: '0x1d9e9118c8ac62360364b659035bd87d110691e7',
      reserve0: '0.095886',
      reserve1: '0.10154694589503153',
      reserveUSD: '0.2260850332568574529014009796904103',
      reserveNativeCurrency: '0.0002011286020137123791924221230501354',
      totalSupply: '0.000000098565255478',
      token0: {
        address: '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48',
        name: 'USD Coin',
        symbol: 'USDC',
        decimals: '6',
      },
      token1: {
        address: '0xd533a949740bb3306d119cc777fa900ba034cd52',
        name: 'Curve DAO Token',
        symbol: 'CRV',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'YMTPP-WETH',
      address: '0x26661b84ec077cd4ce9876657fb35c4d9b0bb0f9',
      reserve0: '673418.539295465611922907',
      reserve1: '0.038159528779165558',
      reserveUSD: '148.8992051270776033411438634668967',
      reserveNativeCurrency: '0.038159528779165558',
      totalSupply: '160.277670087204858882',
      token0: {
        address: '0x61dad5b3c38bb5963b13b325be52ed28ffdf6343',
        name: 'YAMATOPEPE',
        symbol: 'YMTPP',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'NOAHP-WETH',
      address: '0x3ad1754fd7b22af9f9ce66a068e146749aae2d61',
      reserve0: '46235976.082820852285269257',
      reserve1: '0.033801712512721425',
      reserveUSD: '99.66386402218806197538671174414678',
      reserveNativeCurrency: '0.033801712512721425',
      totalSupply: '1241.099995971315760731',
      token0: {
        address: '0x41b3f18c6384dc9a39c33afeca60d9b8e61eaa9f',
        name: 'NOAH DECENTRALIZED STATE COIN',
        symbol: 'NOAHP',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'RAD-WETH',
      address: '0x3fff4a26a98b730c9027d7ab70a2e89916cacd52',
      reserve0: '0.172037299461729311',
      reserve1: '0.000545603308030155',
      reserveUSD: '3.400794002132388789368633737174888',
      reserveNativeCurrency: '0.00109120661606031',
      totalSupply: '0.009599385258801677',
      token0: {
        address: '0x31c8eacbffdd875c74b94b077895bd78cf1e64a3',
        name: 'Radicle',
        symbol: 'RAD',
        decimals: '18',
      },
      token1: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
    {
      title: 'WETH-PAN',
      address: '0x46575017362add18630dd99246bca8f853b80ede',
      reserve0: '0.001207920729132939',
      reserve1: '47.448993205213036559',
      reserveUSD: '9.379555947938372930418567245075574',
      reserveNativeCurrency: '0.002415841458265878',
      totalSupply: '0.23709933895873348',
      token0: {
        address: '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2',
        name: 'Wrapped Ether',
        symbol: 'WETH',
        decimals: '18',
      },
      token1: {
        address: '0xd56dac73a4d6766464b38ec6d91eb45ce7457c44',
        name: 'Panvala pan',
        symbol: 'PAN',
        decimals: '18',
      },
      liquidityMiningCampaigns: [],
    },
  ],
};
