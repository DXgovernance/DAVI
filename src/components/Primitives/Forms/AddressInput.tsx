import styled from 'styled-components';
import { Input, InputProps } from 'components/Primitives/Forms/Input';
import useENSAvatar from 'hooks/Guilds/ether-swr/ens/useENSAvatar';
import { Avatar } from 'components/Avatar';
import { isAddress, MAINNET_ID } from 'utils';
import { Box } from 'components/Primitives/Layout/Box';
import { FiX } from 'react-icons/fi';

export const ClickableIcon = styled(Box)`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

const AddressInput: React.FC<InputProps<string>> = ({
  value,
  onChange,
  isInvalid,
  disabled = false,
  ...rest
}) => {
  const { imageUrl } = useENSAvatar(value, MAINNET_ID);
  const shouldShowAvatar = !!isAddress(value) || value?.endsWith('.eth');

  return (
    <Input
      {...rest}
      value={value}
      disabled={disabled}
      icon={
        <div>
          {shouldShowAvatar && !isInvalid && (
            <Avatar src={imageUrl} defaultSeed={value} size={24} />
          )}
        </div>
      }
      iconRight={
        <>
          {!disabled && value && (
            <ClickableIcon
              aria-label="clear address"
              onClick={() => onChange('')}
            >
              <FiX size={18} />
            </ClickableIcon>
          )}
        </>
      }
      onChange={e => onChange(e.target.value)}
      isInvalid={isInvalid}
    />
  );
};

export { AddressInput };
