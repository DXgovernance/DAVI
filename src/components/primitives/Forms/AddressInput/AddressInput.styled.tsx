import styled from 'styled-components';
import { Box } from 'components/primitives/Layout/Box';

export const ClickableIcon = styled(Box)`
  display: flex;
  align-items: center;
  cursor: pointer;
`;
