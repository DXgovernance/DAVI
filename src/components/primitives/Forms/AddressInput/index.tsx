export { default as AddressInput } from './AddressInput';
