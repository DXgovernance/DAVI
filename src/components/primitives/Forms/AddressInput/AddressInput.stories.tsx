import { ComponentMeta, ComponentStory } from '@storybook/react';
import { useState } from 'react';
import { ZERO_ADDRESS } from 'utils';
import { AddressInput } from './AddressInput';

export default {
  title: 'AddressInput',
  component: AddressInput,
  argTypes: {
    value: {
      description: 'value',
    },
    onChange: {
      description:
        'method that gets triggered when the value of the input changes',
    },
    isInvalid: {
      description: 'marks the field as invalid, changing its style',
    },
    disabled: {
      description: 'prevents the user from interacting with the input field',
    },
    enableENSName: {
      description:
        'enabled by default, it allows the user to input an ENS name as well as an address. If an ENS name is used, it fetches the corresponding address (or lack thereof)',
    },
  },
} as ComponentMeta<typeof AddressInput>;

const StateWrapper = args => {
  const [value, setValue] = useState(ZERO_ADDRESS);
  return <AddressInput value={value} onChange={setValue} {...args} />;
};

const Template: ComponentStory<typeof AddressInput> = args => (
  <StateWrapper {...args} />
);

export const Simple = Template.bind({});
Simple.args = {};

export const Invalid = Template.bind({});
Invalid.args = { isInvalid: true };

export const Disabled = Template.bind({});
Disabled.args = { disabled: true };
