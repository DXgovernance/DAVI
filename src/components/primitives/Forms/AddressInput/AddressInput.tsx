import { Input } from 'components/primitives/Forms/Input';
import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
import { Avatar } from 'components/Avatar';
import { isAddress, MAINNET_ID } from 'utils';
import { FiX } from 'react-icons/fi';
import useENS from 'hooks/Guilds/ens/useENS';
import { useEffect, useState } from 'react';
import { ClickableIcon } from './AddressInput.styled';
import { AddressInputProps } from './types';

export const AddressInput: React.FC<AddressInputProps> = ({
  value,
  onChange,
  isInvalid,
  disabled = false,
  enableENSName = true,
  ...rest
}) => {
  const { imageUrl } = useENSAvatar(value, MAINNET_ID);
  const shouldShowAvatar = !!isAddress(value);

  const { name: parsedENSName } = useENS(value);

  const [localValue, setLocalValue] = useState(
    (enableENSName && parsedENSName) || value
  );
  const { address: addressFromLocalValue } = useENS(
    localValue.endsWith('.eth') ? localValue : `${localValue}.eth`
  );

  useEffect(() => {
    if (enableENSName && addressFromLocalValue) onChange(addressFromLocalValue);
    else onChange(localValue);
  }, [localValue, enableENSName, addressFromLocalValue, onChange]);

  return (
    <Input
      {...rest}
      value={localValue}
      disabled={disabled}
      data-testid="address input"
      icon={
        <div>
          {shouldShowAvatar && !isInvalid && (
            <Avatar
              src={imageUrl}
              defaultSeed={value}
              size={24}
              aria-label="address avatar"
              data-testid="address avatar"
            />
          )}
        </div>
      }
      iconRight={
        <>
          {!disabled && value && (
            <ClickableIcon
              aria-label="clear address"
              data-testid="clear address"
              onClick={() => setLocalValue('')}
            >
              <FiX size={18} />
            </ClickableIcon>
          )}
        </>
      }
      onChange={e => setLocalValue(e.target.value)}
      isInvalid={isInvalid}
    />
  );
};
