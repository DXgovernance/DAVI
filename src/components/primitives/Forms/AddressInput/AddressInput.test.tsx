import { fireEvent } from '@testing-library/react';
import { MOCK_ADDRESS, MOCK_ENS_NAME } from 'hooks/Guilds/ens/fixtures';
import { ZERO_ADDRESS } from 'utils';
import { render } from 'utils/tests';
import { AddressInput } from './AddressInput';

jest.mock('hooks/Guilds/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    imageUrl: 'wagmi',
  }),
}));

jest.mock('hooks/Guilds/ens/useENS', () => ({
  __esModule: true,
  default: (value: string) => {
    if (value === MOCK_ENS_NAME || value === MOCK_ADDRESS) {
      return { name: MOCK_ENS_NAME, address: MOCK_ADDRESS };
    } else {
      return { name: null, address: value };
    }
  },
}));

describe(`AddressInput`, () => {
  describe(`General interaction`, () => {
    it(`should clear the input if the X is clicked`, async () => {
      const { findByTestId } = render(
        <AddressInput value={MOCK_ADDRESS} onChange={jest.fn()} />
      );

      const addressField = await findByTestId('address input');
      const clearAddressButton = await findByTestId('clear address');
      fireEvent.click(clearAddressButton);

      expect(addressField).toHaveDisplayValue('');
    });

    it(`should be disabled`, async () => {
      const { findByTestId } = render(
        <AddressInput
          value={MOCK_ADDRESS}
          onChange={jest.fn()}
          disabled={true}
        />
      );

      const addressField = await findByTestId('address input');

      expect(addressField).toBeDisabled();
    });

    it(`should show the avatar if the address is valid`, async () => {
      const { findByTestId } = render(
        <AddressInput
          value={MOCK_ADDRESS}
          onChange={jest.fn()}
          enableENSName={false}
        />
      );

      const addressAvatar = await findByTestId('avatar');

      expect(addressAvatar).toBeInTheDocument();
    });

    it(`should show the avatar if the ENS name is valid`, async () => {
      const { findByTestId } = render(
        <AddressInput value={MOCK_ADDRESS} onChange={jest.fn()} />
      );

      const addressAvatar = await findByTestId('avatar');

      expect(addressAvatar).toBeInTheDocument();
    });

    it(`shouldn't show the avatar if the address is invalid`, async () => {
      const { queryByTestId } = render(
        <AddressInput
          value={'0xinvalidAddress'}
          onChange={jest.fn()}
          enableENSName={false}
        />
      );

      const addressAvatar = queryByTestId('avatar');

      expect(addressAvatar).not.toBeInTheDocument();
    });

    it(`shouldn't show the avatar if the ENS name is invalid`, async () => {
      const { queryByTestId } = render(
        <AddressInput value={'invalidENS.eth'} onChange={jest.fn()} />
      );

      const addressAvatar = queryByTestId('avatar');

      expect(addressAvatar).not.toBeInTheDocument();
    });
  });

  describe(`ENS behaviour`, () => {
    it(`should return an address value if ENSName is ENABLED and inputs an ENS name`, async () => {
      const mockOnChange = jest.fn();

      const { findByTestId } = render(
        <AddressInput value={''} onChange={mockOnChange} />
      );

      const addressField = await findByTestId('address input');
      fireEvent.change(addressField, { target: { value: MOCK_ENS_NAME } });

      expect(mockOnChange).toHaveBeenCalledWith(MOCK_ADDRESS);
    });

    it(`should return an address value if ENSName is ENABLED and inputs an ENS name without it finishing in '.eth'`, async () => {
      const mockOnChange = jest.fn();

      const { findByTestId } = render(
        <AddressInput value={''} onChange={mockOnChange} />
      );

      const addressField = await findByTestId('address input');
      fireEvent.change(addressField, { target: { value: 'wagmi' } });

      expect(mockOnChange).toHaveBeenCalledWith(MOCK_ADDRESS);
    });

    it(`should return null if ENSName is ENABLED and the ENS name doesn't exist`, async () => {
      const mockOnChange = jest.fn();

      const { findByTestId } = render(
        <AddressInput value={''} onChange={mockOnChange} />
      );

      const addressField = await findByTestId('address input');
      fireEvent.change(addressField, { target: { value: 'wrongENS.eth' } });

      expect(mockOnChange).toHaveBeenCalledWith('wrongENS.eth');
    });

    it(`should return an address if ENSName is DISABLED and it inputs an address`, async () => {
      const mockOnChange = jest.fn();

      const { findByTestId } = render(
        <AddressInput
          value={''}
          onChange={mockOnChange}
          enableENSName={false}
        />
      );

      const addressField = await findByTestId('address input');
      fireEvent.change(addressField, { target: { value: MOCK_ADDRESS } });

      expect(mockOnChange).toHaveBeenCalledWith(MOCK_ADDRESS);
    });

    it(`should return the ENS name if ENSName is DISABLED and it inputs an ENS name`, async () => {
      const mockOnChange = jest.fn();

      const { findByTestId } = render(
        <AddressInput
          value={''}
          onChange={mockOnChange}
          enableENSName={false}
        />
      );

      const addressField = await findByTestId('address input');
      fireEvent.change(addressField, { target: { value: MOCK_ENS_NAME } });

      expect(mockOnChange).toHaveBeenCalledWith(MOCK_ENS_NAME);
    });

    it(`should populate the input with the ENS name if the value is an address`, async () => {
      const { findByTestId } = render(
        <AddressInput value={ZERO_ADDRESS} onChange={jest.fn()} />
      );

      const addressField = await findByTestId('address input');

      expect(addressField).toHaveDisplayValue(MOCK_ENS_NAME);
    });

    it(`should populate the input with the address if the value is an address and ENSName is DISABLED`, async () => {
      const { findByTestId } = render(
        <AddressInput
          value={ZERO_ADDRESS}
          onChange={jest.fn()}
          enableENSName={false}
        />
      );

      const addressField = await findByTestId('address input');

      expect(addressField).toHaveDisplayValue(ZERO_ADDRESS);
    });
  });

  describe(`Snapshots`, () => {
    it(`should match snapshot with address`, async () => {
      const { container } = render(
        <AddressInput value={MOCK_ADDRESS} onChange={jest.fn()} />
      );
      expect(container).toMatchSnapshot();
    });

    it(`should match snapshot with ENS disabled`, async () => {
      const { container } = render(
        <AddressInput
          value={MOCK_ADDRESS}
          onChange={jest.fn()}
          enableENSName={false}
        />
      );
      expect(container).toMatchSnapshot();
    });

    it(`should match snapshot isInvalid`, async () => {
      const { container } = render(
        <AddressInput
          value={'0xinvalidAddress'}
          onChange={jest.fn()}
          isInvalid={true}
        />
      );
      expect(container).toMatchSnapshot();
    });
  });
});
