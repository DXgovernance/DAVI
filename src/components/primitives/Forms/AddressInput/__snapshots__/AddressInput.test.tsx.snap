// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressInput Snapshots should match snapshot isInvalid 1`] = `
.c2 {
  border: none;
  background: none;
  background-color: transparent;
  box-shadow: none;
  width: 100%;
  box-sizing: border-box;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #E75C5C;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c0 input {
  color: #E75C5C;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c0 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c0 input::-moz-placeholder {
  color: #A1A6B0;
}

.c0 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c0 input::placeholder {
  color: #A1A6B0;
}

.c0:hover {
  outline: 1px solid #E75C5C;
}

.c0:focus {
  outline: 2px solid #E75C5C;
}

.c1 {
  margin-right: 0.3rem;
}

.c3 {
  margin-left: 0.3rem;
}

.c4 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div />
    </div>
    <input
      aria-label="address input"
      class="c2"
      data-testid="address input"
      value="0xinvalidAddress"
    />
    <div
      class="c3"
    >
      <div
        aria-label="clear address"
        class="c4 c5"
        data-testid="clear address"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`AddressInput Snapshots should match snapshot with ENS disabled 1`] = `
.c4 {
  border: none;
  background: none;
  background-color: transparent;
  box-shadow: none;
  width: 100%;
  box-sizing: border-box;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #303338;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c0 input {
  color: #fff;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c0 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c0 input::-moz-placeholder {
  color: #A1A6B0;
}

.c0 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c0 input::placeholder {
  color: #A1A6B0;
}

.c0:hover {
  outline: 1px solid #fff;
}

.c0:focus {
  outline: 2px solid #fff;
}

.c1 {
  margin-right: 0.3rem;
}

.c5 {
  margin-left: 0.3rem;
}

.c2 {
  height: 1.6rem;
  width: 1.6rem;
}

.c3 {
  border-radius: 50%;
  height: 24px;
  width: 24px;
}

.c6 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <img
          alt="Avatar"
          class="c2 c3"
          data-testid="avatar"
          size="24"
          src="wagmi"
        />
      </div>
    </div>
    <input
      aria-label="address input"
      class="c4"
      data-testid="address input"
      value="0x0000000000000000000000000000000000000000"
    />
    <div
      class="c5"
    >
      <div
        aria-label="clear address"
        class="c6 c7"
        data-testid="clear address"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </div>
    </div>
  </div>
   
</div>
`;

exports[`AddressInput Snapshots should match snapshot with address 1`] = `
.c4 {
  border: none;
  background: none;
  background-color: transparent;
  box-shadow: none;
  width: 100%;
  box-sizing: border-box;
  outline: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  width: 100%;
  border: 1px solid #303338;
  border-radius: 1.5rem;
  padding: 0.5rem 0.8rem;
  background-color: transparent;
}

.c0 input {
  color: #fff;
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 14px;
  font-weight: 500;
  text-align: left;
}

.c0 input::-webkit-input-placeholder {
  color: #A1A6B0;
}

.c0 input::-moz-placeholder {
  color: #A1A6B0;
}

.c0 input:-ms-input-placeholder {
  color: #A1A6B0;
}

.c0 input::placeholder {
  color: #A1A6B0;
}

.c0:hover {
  outline: 1px solid #fff;
}

.c0:focus {
  outline: 2px solid #fff;
}

.c1 {
  margin-right: 0.3rem;
}

.c5 {
  margin-left: 0.3rem;
}

.c2 {
  height: 1.6rem;
  width: 1.6rem;
}

.c3 {
  border-radius: 50%;
  height: 24px;
  width: 24px;
}

.c6 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div>
        <img
          alt="Avatar"
          class="c2 c3"
          data-testid="avatar"
          size="24"
          src="wagmi"
        />
      </div>
    </div>
    <input
      aria-label="address input"
      class="c4"
      data-testid="address input"
      value="wagmi.eth"
    />
    <div
      class="c5"
    >
      <div
        aria-label="clear address"
        class="c6 c7"
        data-testid="clear address"
      >
        <svg
          fill="none"
          height="18"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="18"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="18"
            x2="6"
            y1="6"
            y2="18"
          />
          <line
            x1="6"
            x2="18"
            y1="6"
            y2="18"
          />
        </svg>
      </div>
    </div>
  </div>
   
</div>
`;
